app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -167,6 +169,18 @@ def get_managed_images
         process_collection(images, :vms) { |image| parse_managed_image(image) }
       end
 
+      def get_market_images
+        urns   = @options.market_image_urns
+        images = if urns
+                   gather_data_for_this_region(@vmis).select do |image|
+                     urns.include?(image.id)
+                   end
+                 else
+                   gather_data_for_this_region(@vmis)
+                 end
```

Comment:
We do the same thing for Amazon, honestly.  If the user enables public images and does not set a filter on which public images to pull in, they're opening themselves up for 10k+ images.

And, yeah, it has a performance impact in inventory collection, especially since we gather them each time we do inventory collection to do the inventory comparisons.

There is work underway in inventory collection optimizations that actually improve this situation quite a bit (for those that actually want to see *all* those images).  But, that's a ways off.

I'd say for now, since we do it already for another provider (and arguably, more than double the images), it's probably OK here.

Maybe just add a simple comment to the settings.yml file indicating that neglecting to set a filter and enabling marketplace images will result in much slower inventory collection times, and it's advisable to set a marketplace image filter.